Mathlib/Topology/Algebra/Algebra.lean

```diff
@@ -74,16 +78,471 @@ theorem algebraMapCLM_toLinearMap : (algebraMapCLM R A).toLinearMap = Algebra.li
+  is also a topological `R`-algebra. -/
+instance DiscreteTopology.continuousSMul [TopologicalSemiring A] [DiscreteTopology R] :
+    ContinuousSMul R A := continuousSMul_of_algebraMap _ _ continuous_of_discreteTopology
+
 end TopologicalAlgebra
 
-section TopologicalAlgebra
+section TopologicalAlgebra --
+section
 
 variable {R : Type*} [CommSemiring R]
 variable {A : Type u} [TopologicalSpace A]
 variable [Semiring A] [Algebra R A]
 
 #align subalgebra.has_continuous_smul SMulMemClass.continuousSMul
 
+end
+section
+
+variable (R : Type*) [CommSemiring R] [TopologicalSpace R] [TopologicalSemiring R]
+  (A : Type*) [Semiring A] [TopologicalSpace A] [TopologicalSemiring A]
+
+/-- Continuous algebra homomorphisms between algebras. We only put the type classes that are
+  necessary for the definition, although in applications `M` and `B` will be topological algebras
+  over the topological ring `R`. -/
+structure ContinuousAlgHom (R : Type*) [CommSemiring R] (A : Type*) [Semiring A]
+    [TopologicalSpace A] (B : Type*) [Semiring B] [TopologicalSpace B] [Algebra R A] [Algebra R B]
+    extends A →ₐ[R] B where
+  cont : Continuous toFun := by continuity
+
+/-- `ContinuousAlgHomClass F R A B` asserts `F` is a type of bundled continuous `R`-agebra
+  homomorphisms `A → B`. -/
+class ContinuousAlgHomClass (F : Type*) (R : outParam (Type*)) [CommSemiring R]
+    (A : outParam (Type*)) [Semiring A] [TopologicalSpace A]
+    (B : outParam (Type*)) [Semiring B] [TopologicalSpace B][Algebra R A]
+    [Algebra R B] [FunLike F A B]
+    extends AlgHomClass F R A B, ContinuousMapClass F A B : Prop
+attribute [inherit_doc ContinuousAlgHom] ContinuousAlgHom.cont
+
+@[inherit_doc]
+notation:25 A " →A[" R "] " B => ContinuousAlgHom R A B
+
+namespace ContinuousAlgHom
+
+section Semiring
+
+variable {R} {A}
+variable [TopologicalSpace R] [TopologicalSemiring R]
+
+variable {B : Type*} [Semiring B] [TopologicalSpace B] [Algebra R A] [Algebra R B]
+
+attribute [coe] ContinuousAlgHom.toAlgHom
+/-- Coerce continuous algebra morphisms to algebra morphisms. -/
+instance AlgHom.coe : Coe (A →A[R] B) (A →ₐ[R] B) := ⟨toAlgHom⟩
+
+theorem coe_injective : Function.Injective ((↑) : (A →A[R] B) → A →ₐ[R] B) := by
+  intro f g H
+  cases f
+  cases g
+  congr
+
+instance funLike : FunLike (A →A[R] B) A B where
+  coe f := f.toAlgHom
+  coe_injective'  _ _ h  := coe_injective (DFunLike.coe_injective h)
+
+instance continuousAlgHomClass :
+    ContinuousAlgHomClass (A →A[R] B) R A B where
+  map_mul f x y    := map_mul f.toAlgHom x y
+  map_one f        := map_one f.toAlgHom
+  map_add f        := map_add f.toAlgHom
+  map_zero f       := map_zero f.toAlgHom
+  commutes f r     := f.toAlgHom.commutes r
+  map_continuous f := f.2
+
+theorem coe_mk (f : A →ₐ[R] B) (h) : (mk f h : A →ₐ[R] B) = f := rfl
+
+@[simp]
+theorem coe_mk' (f : A →ₐ[R] B) (h) : (mk f h : A → B) = f := rfl
+
+@[continuity]
```

Comment:
This will be the default soon, and hopefully we can delete `continuity`. If you still need the `continuity` tag for now, that's fine, just include both in that case.
```suggestion
@[fun_prop]
```
Also, it shouldn't actually be necessary, as it should just follow from the `ContinuousMapClass` instance, but I think having the explicit lemma does help `fun_prop`'s performance.